.circleci/config.yml

```diff
@@ -665,7 +665,7 @@ jobs:
             TEST_FILES=$(<<parameters.test_list>>)
             if [ -z "$TEST_FILES" ]; then
               echo "No test files to run. Exiting early."
-              circleci-agent step halt
+              exit 1
```

Comment:
Could you explain the reason to replace `circleci-agent step halt` with `exit 1`?

Reply:
The reason to replace `circleci-agent step halt` with `exit 1` is because `circleci-agent` is no longer available since we migrated the job from container runner to machine runner. 
That being said, it might make more sense to use exit code 0 to indicate that the job exited gracefully.